### DISCUSSION TITLE
Internal Error: PHP extension required: "pdo_sqlite"

### DISCUSSION THREAD

**Discussion by githubIMC** [kanboard/kanboard#5371] [Installation & Configuration]
I use mariadb (mysql  Ver 15.1 Distrib 10.5.19-MariaDB,) and PHP 8.2
when I try to install I have an error and installation php-sqlite3 is ok

**Comment by fguillot** [comment]
See: https://github.com/kanboard/kanboard/blob/09f38f072a541bc6049345568d1095633ff7870e/app/check_setup.php#L14-L20

**Reply by anonim-01** [reply]
```
if (DB_DRIVER === 'sqlite' && !extension_loaded('pdo_sqlite')) { 
    throw new Exception('PHP extension required: "pdo_sqlite"'); 
} 

if (DB_DRIVER === 'mySQL' && !extension_loaded('pdo_mysql')) { 
    throw new Exception('PHP extension required: "pdo_mysql"'); 
}

```


> 
> The purpose of this code is to throw an exception based on the database driver and the required PDO extensions, displaying an error message when these extensions are missing. As a solution, you can follow these steps to resolve these errors:
> 
> 1. Install pdo_sqlite or pdo_mysql Extensions:
>    - If you are using an SQLite database, you need to install the pdo_sqlite extension.
>    - If you are using a MySQL database, you need to install the pdo_mysql extension.
>    - After installing the relevant extensions, your code should not encounter these errors.
> 
> 2. Set the Database Driver Correctly:
>    - Ensure that the DB_DRIVER variable has the correct value. It should be 'sqlite' or 'mySQL' in your code, and pay attention to case sensitivity.
> 
> If you are still experiencing issues despite following these steps, you can share more details for further assistance. Details such as the PHP version you are using and the operating system can be helpful.
> The purpose of this code is to throw an exception based on the database driver and the required PDO extensions, displaying an error message when these extensions are missing. As a solution, you can follow these steps to resolve these errors:
> 
> 1. Install pdo_sqlite or pdo_mysql Extensions:
>    - If you are using an SQLite database, you need to install the pdo_sqlite extension.
>    - If you are using a MySQL database, you need to install the pdo_mysql extension.
>    - After installing the relevant extensions, your code should not encounter these errors.
> 
> 2. Set the Database Driver Correctly:
>    - Ensure that the DB_DRIVER variable has the correct value. It should be 'sqlite' or 'mySQL' in your code, and pay attention to case sensitivity.
> 
> If you are still experiencing issues despite following these steps, you can share more details for further assistance. Details such as the PHP version you are using and the operating system can be helpful.


Solution:

If you encounter these errors locally, you can follow these steps:

Check PDO Extensions:

If you are using SQLite, make sure the extension=php_pdo_sqlite.dll line is enabled in your php.ini file.
If you are using MySQL, make sure the extension=php_pdo_mysql.dll line is enabled in your php.ini file.
Check DB_DRIVER Value:

Check the DB_DRIVER variable in your code that determines the database driver. If you are using MySQL, the value should be set to mySQL (pay attention to case sensitivity).
Install PDO Extensions:

If you still encounter errors after the above checks, install the specified PDO extensions on your computer. You can obtain these extensions from the [official PHP website](https://www.php.net/manual/en/pdo.installation.php) or from the documentation of the local server you are using.
Restart the Web Server:

Restart the web server (usually Apache or Nginx) you are using to apply the changes.
Check PHP Error Logs in Case of Failure:

If you still encounter an error, check the PHP error logs. These errors are usually located in the log directory of the web server or in a specific location. You may find a more specific error message in the error logs.
If the problem persists after following these steps, you can share more details such as the web server and PHP version you are using for further assistance.

**Comment by anonim-01** [comment]
> I use mariadb (mysql Ver 15.1 Distrib 10.5.19-MariaDB,) and PHP 8.2 when I try to install I have an error and installation php-sqlite3 is ok

MySQL and MariaDB have different PHP extensions for database connectivity. MariaDB generally uses the `mysqli` extension, while SQLite uses `sqlite3`. The error you're encountering suggests that the required extension for MySQL (probably `pdo_mysql`) might not be installed or properly configured.

Here are some steps you can take to resolve the issue:

1. **Check if `pdo_mysql` Extension is Installed:**
   Make sure that the `pdo_mysql` extension is installed. You can do this by checking your `php.ini` file or by running the following command in your terminal or command prompt:
   ```bash
   php -m
   ```
   Look for `pdo_mysql` in the list of installed modules.

2. **Install `pdo_mysql` Extension:**
   If it's not installed, you can install it using your package manager. For example, if you're using Ubuntu, you can run:
   ```bash
   sudo apt-get install php8.2-mysql
   ```
   Adjust the command based on your operating system.

3. **Check `extension_dir` in `php.ini`:**
   Ensure that the `extension_dir` directive in your `php.ini` file is set correctly. This should point to the directory where PHP extensions are located. For example:
   ```ini
   extension_dir = "/usr/lib/php/20170718/"
   ```

4. **Restart Your Web Server:**
   After making changes to the `php.ini` file, restart your web server to apply the changes:
   ```bash
   sudo service apache2 restart
   ```
   Substitute `apache2` with your actual web server service name.

5. **Check for Errors:**
   Check your web server error logs for any PHP-related errors. They can provide more details about what's going wrong.

6. **Verify Database Connection Configuration:**
   Double-check your database connection configuration in your PHP code. Ensure that the host, username, password, and database name are correctly set.

If the issue persists, please provide more details about the error message you are receiving during the installation or any relevant error logs. This information will help in providing more targeted assistance.

**Reply by githubIMC** [reply]
i have check the pdo_mysql is on:
_PDO
PDO support | enabled
-- | --
PDO drivers | mysql_

the problem that the request is not on pdo_mysql but on pdo_sqlite !!!
